tests/unit/test_s3_orc.py

```diff
@@ -0,0 +1,535 @@
+    assert list(df2.columns) == ["c0", "camelcase", "c_2"]
+    assert df2.c0.sum() == 1
+    assert df2.camelcase.sum() == 5
+    assert df2.c_2.sum() == 9
+
+
+@pytest.mark.parametrize("use_threads", [True, False, 2])
+def test_to_orc_file_dtype(path, use_threads):
+    df = pd.DataFrame({"c0": [1.0, None, 2.0], "c1": [pd.NA, pd.NA, pd.NA]})
+    file_path = f"{path}0.orc"
+    wr.s3.to_orc(df, file_path, dtype={"c0": "bigint", "c1": "string"}, use_threads=use_threads)
+    df2 = wr.s3.read_orc(file_path, use_threads=use_threads)
+    assert df2.shape == df.shape
+    assert df2.c0.sum() == 3
+    assert str(df2.c0.dtype) == "Int64"
+    assert str(df2.c1.dtype) == "string"
+
+
+@pytest.mark.parametrize("filename_prefix", [None, "my_prefix"])
+@pytest.mark.parametrize("use_threads", [True, False])
+def test_to_orc_filename_prefix(compare_filename_prefix, path, filename_prefix, use_threads):
+    test_prefix = "my_prefix"
+    df = pd.DataFrame({"col": [1, 2, 3], "col2": ["A", "A", "B"]})
+    file_path = f"{path}0.orc"
+
+    # If Dataset is False, ORC file should never start with prefix
+    filename = wr.s3.to_orc(
+        df=df, path=file_path, dataset=False, filename_prefix=filename_prefix, use_threads=use_threads
+    )["paths"][0].split("/")[-1]
+    assert not filename.startswith(test_prefix)
+
+    # If Dataset is True, ORC file starts with prefix if one is supplied
+    filename = wr.s3.to_orc(df=df, path=path, dataset=True, filename_prefix=filename_prefix, use_threads=use_threads)[
+        "paths"
+    ][0].split("/")[-1]
+    compare_filename_prefix(filename, filename_prefix, test_prefix)
+
+    # Partitioned
+    filename = wr.s3.to_orc(
+        df=df,
+        path=path,
+        dataset=True,
+        filename_prefix=filename_prefix,
+        partition_cols=["col2"],
+        use_threads=use_threads,
+    )["paths"][0].split("/")[-1]
+    compare_filename_prefix(filename, filename_prefix, test_prefix)
+
+    # Bucketing
+    filename = wr.s3.to_orc(
+        df=df,
+        path=path,
+        dataset=True,
+        filename_prefix=filename_prefix,
+        bucketing_info=(["col2"], 2),
+        use_threads=use_threads,
+    )["paths"][0].split("/")[-1]
+    compare_filename_prefix(filename, filename_prefix, test_prefix)
+    assert filename.endswith("bucket-00000.orc")
+
+
+def test_read_orc_map_types(path):
+    df = pd.DataFrame({"c0": [0, 1, 1, 2]}, dtype=np.int8)
+    file_path = f"{path}0.orc"
+    wr.s3.to_orc(df, file_path)
+    df2 = wr.s3.read_orc(file_path)
+    assert str(df2.c0.dtype) == "Int8"
+    df3 = wr.s3.read_orc(file_path, pyarrow_additional_kwargs={"types_mapper": None})
+    assert str(df3.c0.dtype) == "int8"
+
+
+@pytest.mark.parametrize("use_threads", [True, False, 2])
+@pytest.mark.parametrize("max_rows_by_file", [None, 0, 40, 250, 1000])
+def test_orc_with_size(path, use_threads, max_rows_by_file):
+    df = get_df_list().drop(["category"], axis=1)  # category not supported
```

Comment:
So ORC does not know how to work with categoricals? I think even for parquet tables they are converted into "dictionary arrays" when creating arrow tables. Does it fail to write that into ORC?

Reply:
The issue is that PyArrow ORC doesn't support the dictionary arrays. When I have a category type, this error pops up:
```
pyarrow.lib.ArrowNotImplementedError: Unknown or unsupported Arrow type: dictionary<values=double, indices=int8, ordered=0>
```

Reply:
Ah it's a shame. Ok, thanks!